src/vs/editor/contrib/rtv/RTVSynthService.ts

```diff
@@ -0,0 +1,327 @@
+import { TableElement, isHtmlEscape } from 'vs/editor/contrib/rtv/RTVUtils';
+
+export class RTVSynthService {
+	private _allEnvs: any[] = [];
+	private _prevEnvs?: Map<number, any>;
+	private _boxEnvs: any[] = [];
+	private _boxVars: Set<string> = new Set<string>();
+	private _lineNumber: number;
+	private _rowsValid: boolean[] = [];
+	private _includedTimes: Set<number> = new Set();
+	private _outputVars: string[];
+	private _rows?: TableElement[][];
+	private onBoxContentChanged?: Function;
+
+	constructor(
+		outputVars: string[],
+		lineno: number,
+		boxVars: Set<string>
+	) {
+		this._outputVars = outputVars;
+		this._lineNumber = lineno;
+		this._boxVars = boxVars;
+	}
+
+	get boxEnvs(): any[] {
+		return this._boxEnvs;
+	}
+
+	get includedTimes(): Set<number> {
+		return this._includedTimes;
+	}
+
+	get prevEnvs(): Map<number, any> {
+		return this._prevEnvs!;
+	}
+
+	get varnames(): string[] {
+		return this._outputVars!;
+	}
+
+	bindBoxContentChanged(callback: Function) {
```

Comment:
Can we switch the `Function` type to the type that includes the parameter(s) and return type? I think the syntax is `(parameter1: type1, parameter2: type2, ...) => returntype`.